examples/maturin_extension/maturin_extension/register_submodules.py

```diff
@@ -0,0 +1,20 @@
+import os
+import sys
+import importlib
+import importlib.abc
+
+# Import the local extension
+from .maturin_extension import __file__ as rust_extension_path
+
+
+class SubmoduleFinder(importlib.abc.MetaPathFinder):
```

Comment:
huh what's that doing? I wasn't aware that maturin needs that kind of workarounds

Reply:
It's to enable importing directly from a submodule.

E.g. this always works:

```python
import maturin_extension
maturin_extension.datetime.foo()
```

But the following only works with the `SubmoduleFinder`

```python
import maturin_extension.datetime
datetime.foo()
# or
from maturin_extension.datetime import foo
foo()
```

However I'm beginning to think this was perhaps not a good idea: it re-runs the `#[pymodule]` init code which means that it's possible to create two separate modules:

```python
>>> import maturin_extension
>>> dt = maturin_extension.datetime
>>> import maturin_extension.datetime
>>> dt == maturin_extension.datetime
False
```

That it's possible to create two different copies of the submodule is a bit... horrible. So I'm going to remove this from the example code.